src/data/set/basic.lean

```diff
@@ -1817,6 +1820,24 @@ lemma range_diff_image {f : α → β} (H : injective f) (s : set α) :
 subset.antisymm (range_diff_image_subset f s) $ λ y ⟨x, hx, hy⟩, hy ▸
   ⟨mem_range_self _, λ ⟨x', hx', eq⟩, hx $ H eq ▸ hx'⟩
 
+/-- We can use the axiom of choice to pick a preimage for every element of `range f`. -/
+noncomputable def range_splitting (f : α → β) : range f → α := λ x, x.2.some
+
+lemma apply_range_splitting (f : α → β) (x : range f) : f (range_splitting f x) = x :=
```

Comment:
should this be `@[simp]`? Otherwise LGTM

Reply:
This can't be a `@[simp]` lemma, because the head symbol is a variable. (I made the same mistake, until the linter reminded me of this.)